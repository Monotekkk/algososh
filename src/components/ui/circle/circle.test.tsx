import React from 'react';
import renderer from 'react-test-renderer';
import {Circle} from "./circle";
import {ElementStates} from "../../../types/element-states";

it('circle without letter', () => {
    const tree = renderer
        .create(<Circle />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('circle with letter', () => {
    const tree = renderer
        .create(<Circle />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('circle with head', () => {
    const tree = renderer
        .create(<Circle head={'test text'}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('circle with react in head', () => {
    const tree = renderer
        .create(<Circle head={<Circle />} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('circle with tail', () => {
    const tree = renderer
        .create(<Circle tail={'test text'}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('circle with react in tail', () => {
    const tree = renderer
        .create(<Circle tail={<Circle />} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('circle with index', () => {
    const tree = renderer
        .create(<Circle index={0} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('circle with props isSmall ===  true', () => {
    const tree = renderer
        .create(<Circle isSmall={true} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('circle with state in default', () => {
    const tree = renderer
        .create(<Circle state={ElementStates.Default} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('circle with state in changing', () => {
    const tree = renderer
        .create(<Circle state={ElementStates.Changing} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
it('circle with state in modified', () => {
    const tree = renderer
        .create(<Circle state={ElementStates.Modified} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});