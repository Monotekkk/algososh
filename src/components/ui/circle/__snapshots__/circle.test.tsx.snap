// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    test text
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`circle with letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with letter 2`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with out letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with props 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`circle with props isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with react in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with react in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`circle with state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with state in changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with state in changing 2`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with state in default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with state in default 2`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with state in modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    test text
  </div>
</div>
`;

exports[`circle without letter 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle с head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    test text
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    test text
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
